src/plugins/intel_cpu/src/nodes/rdft.cpp

```diff
@@ -164,21 +165,28 @@ void RDFT::execute(dnnl::stream strm) {
 
     auto rank = inputShape.size() - inverse;
 
-    if (axes.size() == 0) {
+    if (!isAxesConstant) {
```

Comment:
This solution, although being applicable, is not that effective. The thing is that the input data on the `axis` and `singnal_sizes` do not change every infer request, that means can be reused for the previous one, even though the input parameters are variables. This situation is quite usual for node with dynamic shapes support and the CPU plugin base classes help to optimize such application. There are three additional interface calls in the `Node` base class that might be usefull here.
1. `prepareParams` this operation is called every infer request if the node has dynamic shapes. So all the shape dependent algorithm/executor preparations should be moved to this stage.
2. `needPrepareParams` this call return a status on wether we should call `prepareParams` on this particular infer request, or not. Usually, inside this call, checks are performed on the input variable parameters and the input shapes. See `Gather::needPrepareParams()` for example.
3. If the shape inference result depends on the input parameters, it's worth to add the same check for the input values change in the `needShapeInfer` call, so that recuse the amount of shape inference calls. However, it does make sense only when the shape infer call is slower than check itself. See `Deconvolution::needShapeInfer()` or `Broadcast::needShapeInfer()` as an example.

Reply:
ok, so I have implemented prepareParams, needPrepareParams and needShapeInfer. Can you rereview?